app/core/Tracker2/SegmentTracker.js

```diff
@@ -103,49 +114,92 @@ export default class SegmentTracker extends BaseTracker {
   }
 
   async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
     this.store.watch(
       (state, getters) => getters['me/isTeacher'],
-      this.onIsTeacherChanged.bind(this)
+      () => this.onIsTeacherChanged()
     )
 
     if (this.store.getters['me/isTeacher']) {
-      loadSegment()
+      this.onIsTeacherChanged(true)
       window.analytics.ready(this.onInitializeSuccess)
     } else {
       this.onInitializeSuccess()
     }
   }
 
-  async trackPageView (includeIntegrations = {}) {
+  async trackPageView (includeIntegrations = []) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
 
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    this.addIntegrationsToSegmentOptions(options, includeIntegrations)
+
     const url = `/${Backbone.history.getFragment()}`
-    window.analytics.page(url, {}, {
-      integrations: {
-        All: false,
-        'Google Analytics': false
-      }
+    return new Promise((resolve) => {
+      window.analytics.page(undefined, url, {}, options, resolve)
     })
   }
 
+  // TODO legacy tracker filtered out identify calls when me.get('unsubscribedFromMarketingEmails') is true
```

Comment:
Is this a filter we should defensively keep on until we have migrated that data into our third party services?

Reply:
This is related to https://github.com/codecombat/codecombat/pull/5797/files#r395281367, I don't think we need to because customer.io wont' be enabled until we have subscription status properly set.

Reply:
This was added very early on when I was extracting segment logic from tracker. Don't think we need it anymore!